src/pkg/services/m365/api/channels_pager.go

```diff
@@ -141,6 +152,109 @@ func (c Channels) GetChannelMessageIDsDelta(
 	return vs, du, nil
 }
 
+// ---------------------------------------------------------------------------
+// channel message replies pager
+// ---------------------------------------------------------------------------
+
+var _ Pager[models.ChatMessageable] = &channelMessageRepliesPageCtrl{}
+
+type channelMessageRepliesPageCtrl struct {
+	gs      graph.Servicer
+	builder *teams.ItemChannelsItemMessagesItemRepliesRequestBuilder
+	options *teams.ItemChannelsItemMessagesItemRepliesRequestBuilderGetRequestConfiguration
+}
+
+func (p *channelMessageRepliesPageCtrl) SetNext(nextLink string) {
+	p.builder = teams.NewItemChannelsItemMessagesItemRepliesRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *channelMessageRepliesPageCtrl) GetPage(
+	ctx context.Context,
+) (PageLinker, error) {
+	resp, err := p.builder.Get(ctx, p.options)
+	return resp, graph.Stack(ctx, err).OrNil()
+}
+
+func (p *channelMessageRepliesPageCtrl) GetOdataNextLink() *string {
+	return ptr.To("")
+}
+
+func (p *channelMessageRepliesPageCtrl) ValuesIn(l PageLinker) ([]models.ChatMessageable, error) {
+	return getValues[models.ChatMessageable](l)
+}
+
+func (c Channels) NewChannelMessageRepliesPager(
+	teamID, channelID, messageID string,
+	selectProps ...string,
+) *channelMessageRepliesPageCtrl {
+	options := &teams.ItemChannelsItemMessagesItemRepliesRequestBuilderGetRequestConfiguration{
+		Headers: newPreferHeaders(preferPageSize(maxNonDeltaPageSize)),
+	}
+
+	if len(selectProps) > 0 {
+		options.QueryParameters.Select = selectProps
+	}
+
+	res := &channelMessageRepliesPageCtrl{
+		gs:      c.Stable,
+		options: options,
+		builder: c.Stable.
+			Client().
+			Teams().
+			ByTeamId(teamID).
+			Channels().
+			ByChannelId(channelID).
+			Messages().
+			ByChatMessageId(messageID).
+			Replies(),
+	}
+
+	return res
+}
+
+// GetChannels fetches the minimum valuable data from each reply in the message
+func (c Channels) GetChannelMessageReplies(
+	ctx context.Context,
+	teamID, channelID, messageID string,
+) ([]models.ChatMessageable, error) {
+	var (
+		vs = []models.ChatMessageable{}
+		// select is not currently enabled for replies.
+		// selectProps = idAnd(
+		// 	"messageType",
+		// 	"createdDateTime",
+		// 	"from",
+		// 	"body")
+		pager = c.NewChannelMessageRepliesPager(teamID, channelID, messageID)
+	)
+
+	// Loop through all pages returned by Graph API.
+	for {
+		page, err := pager.GetPage(ctx)
+		if err != nil {
+			return nil, graph.Wrap(ctx, err, "retrieving page of channels")
+		}
+
+		vals, err := pager.ValuesIn(page)
+		if err != nil {
+			return nil, graph.Wrap(ctx, err, "extracting channels from response")
+		}
+
+		vs = append(vs, vals...)
+
+		nextLink := ptr.Val(page.GetOdataNextLink())
+		if len(nextLink) == 0 {
+			break
+		}
+
+		pager.SetNext(nextLink)
+	}
+
+	logger.Ctx(ctx).Debugf("retrieved %d channel message replies", len(vs))
```

Comment:
nit: stray debug log?
```suggestion
```

Reply:
Not stray.  Could be an info, i suppose.